solana-programs/audius_eth_registry/src/processor.rs

```diff
@@ -132,8 +329,267 @@ impl Processor {
+        // Sysvar Instruction account info
+        let instruction_info = next_account_info(account_info_iter)?;
+
+        // Confirm program ownership of SignerGroup and ValidSigners
+        if valid_signer_1_info.owner != _program_id
+            || valid_signer_2_info.owner != _program_id
+            || valid_signer_3_info.owner != _program_id
+            || old_valid_signer_info.owner != _program_id
+            || signer_group_info.owner != _program_id {
+            return Err(AudiusError::InvalidInstruction.into());
+        }
+
+        // clock sysvar account
+        let clock_account_info = next_account_info(account_info_iter)?;
+        let clock = Clock::from_account_info(&clock_account_info)?;
+
+        // Recover all secp instructions present in this tx
+        let instruction_recovery = Self::recover_secp_instructions(&instruction_info);
+        if instruction_recovery.is_err() {
+            return Err(AudiusError::Secp256InstructionLosing.into());
+        }
+
+        let recovered_instructions = instruction_recovery.unwrap();
+        if recovered_instructions.len() < 3 {
+            return Err(AudiusError::Secp256InstructionLosing.into());
+        }
+
+        let secp_instruction_1 = &recovered_instructions[0];
+        let secp_instruction_2 = &recovered_instructions[1];
+        let secp_instruction_3 = &recovered_instructions[2];
+
+        let valid_signer_1 = Box::new(ValidSigner::try_from_slice(
+            &valid_signer_1_info.data.borrow(),
+        )?);
+        let valid_signer_2 = Box::new(ValidSigner::try_from_slice(
+            &valid_signer_2_info.data.borrow(),
+        )?);
+        let valid_signer_3 = Box::new(ValidSigner::try_from_slice(
+            &valid_signer_3_info.data.borrow(),
+        )?);
+
+        if !valid_signer_1.is_initialized()
+            || !valid_signer_2.is_initialized()
+            || !valid_signer_3.is_initialized()
+        {
+            return Err(AudiusError::ValidSignerNotInitialized.into());
+        }
+
+        if valid_signer_1.signer_group != *signer_group_info.key
+            || valid_signer_2.signer_group != *signer_group_info.key
+            || valid_signer_3.signer_group != *signer_group_info.key
+        {
+            return Err(AudiusError::WrongSignerGroup.into());
+        }
+
+        let instruction_data_1 = Self::recover_instruction_data(&signature_data_1, &valid_signer_1);
+        let instruction_data_2 = Self::recover_instruction_data(&signature_data_2, &valid_signer_2);
+        let instruction_data_3 = Self::recover_instruction_data(&signature_data_3, &valid_signer_3);
+
+        if instruction_data_1 != secp_instruction_1.data {
+            return Err(AudiusError::SignatureVerificationFailed.into());
+        }
+
+        if instruction_data_2 != secp_instruction_2.data {
+            return Err(AudiusError::SignatureVerificationFailed.into());
+        }
+
+        if instruction_data_3 != secp_instruction_3.data {
```

Comment:
I do not see if there is a check for the instruction to go to secp verification program. Without this check it is possible to supply arbitrary instruction calls here and fool this program into believing there is an actual signature verification.

Reply:
https://github.com/atticwip/claimable-tokens/blob/8256bd04a1aaaa9fc405beee3fd703a66526f2a4/src/processor.rs#L196